hypothesis-python/src/hypothesis/internal/conjecture/datatree.py

```diff
@@ -333,25 +422,37 @@ def draw_bits(self, n_bits, forced, value):
                 node.split_at(i)
                 assert i == len(node.values)
                 new_node = TreeNode()
-                branch = node.transition
-                branch.children[value] = new_node
+                node.transition.children[value] = new_node
                 self.__current_node = new_node
                 self.__index_in_current_node = 0
         else:
             trans = node.transition
             if trans is None:
-                node.bit_lengths.append(n_bits)
+                node.ir_types.append(ir_type)
+                node.kwargs.append(kwargs)
                 node.values.append(value)
                 if forced:
                     node.mark_forced(i)
+                # generate_novel_prefix assumes the following invariant: any one
+                # of the series of draws in a particular node can vary. This is
+                # true if all nodes have more than one possibility, which was
+                # true when the underlying representation was bits (lowest was
+                # n=1 bits with m=2 choices).
+                # However, with the ir, e.g. integers(0, 0) has only a single
+                # value. To retain the invariant, we forcefully split such cases
+                # into a transition.
+                if compute_max_children(kwargs, ir_type) == 1:
+                    node.split_at(i)
+                    self.__current_node = node.transition.children[value]
+                    self.__index_in_current_node = 0
```

Comment:
worth pointing out this comment. I'm relatively confident in the assessment here, but maybe there's a more elegant way to handle this.